packages/ai-terminal/src/browser/ai-terminal-agent.ts

```diff
@@ -181,12 +199,28 @@ recent-terminal-contents:
                 // model returned structured output
                 const parsedResult = Commands.safeParse(result.parsed);
                 if (parsedResult.success) {
+                    const responseText = JSON.stringify(parsedResult.data.commands);
+                    this.recordingService.recordResponse({
+                        agentId: this.id,
+                        sessionId,
+                        timestamp: Date.now(),
+                        requestId,
+                        response: responseText,
+                    });
                     return parsedResult.data.commands;
                 }
             }
 
             // fall back to agent-based parsing of result
             const jsonResult = await getJsonOfResponse(result);
+            const responseText = JSON.stringify(jsonResult);
+            this.recordingService.recordResponse({
+                agentId: this.id,
+                sessionId,
+                timestamp: Date.now(),
+                requestId,
+                response: responseText
+            });
```

Comment:
```suggestion
            this.recordingService.recordResponse({
                agentId: this.id,
                sessionId,
                timestamp: Date.now(),
                requestId,
                response: JSON.stringify(jsonResult)
            });
```

Reply:
I had this before, but especially with GPT 3.5 it "hanged" sometimes. I guess the error in parsing is not thrown correctly then, could this be?

Reply:
I fixed the linting error with the new commit